pkg/querier/stats/stats_test.go

```diff
@@ -170,6 +175,10 @@ func TestStats_Merge(t *testing.T) {
 		stats2.AddSplitQueries(11)
 		stats2.AddQueueTime(10 * time.Second)
 		stats2.AddSamplesProcessed(20)
+		stats2.AddSamplesProcessedPerStep([]StepStat{
```

Comment:
in this test case it's not clear what we're testing  - that the stats are deduplicated or that we pick the first or the last or just any of them